internal/rpm/files_test.go

```diff
@@ -0,0 +1,128 @@
+package rpm
+
+import (
+	"context"
+	"runtime"
+	"testing"
+	"unique"
+	"weak"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/test"
+)
+
+type FileCheckerFileTestcase struct {
+	Name  string
+	Owned bool
+	Path  string
+}
+
+type FileCheckerTestcase struct {
+	Name      string
+	Layer     test.LayerRef
+	Testcases []FileCheckerFileTestcase
+}
+
+func (tc *FileCheckerTestcase) Run(ctx context.Context, t *testing.T, a *test.CachedArena) {
+	t.Run(tc.Name, func(t *testing.T) {
+		ctx := zlog.Test(ctx, t)
+		a.LoadLayerFromRegistry(ctx, t, tc.Layer)
+		r := a.Realizer(ctx).(*test.CachedRealizer)
+		t.Cleanup(func() {
+			if err := r.Close(); err != nil {
+				t.Error(err)
+			}
+		})
+
+		rl, err := r.RealizeDescriptions(ctx, []claircore.LayerDescription{
+			{
+				Digest:    tc.Layer.Digest,
+				URI:       "http://example.com",
+				MediaType: test.MediaType,
+				Headers:   make(map[string][]string),
+			},
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		set, err := NewPathSet(ctx, &rl[0])
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer runtime.KeepAlive(set)
+
+		for _, tc := range tc.Testcases {
+			t.Run(tc.Name, func(t *testing.T) {
+				t.Parallel()
```

Comment:
https://go.dev/doc/gc-guide#Testing_object_death suggests avoiding testing object death (perhaps also things related to object death) in parallel with other tests. Should we remove this line?

Reply:
It's not testing object death, it's just ensuring the cache is used.